ui-src/src/app/test-suites/test-suites.component.ts

```diff
@@ -1,8 +1,166 @@
-import { Component, OnInit } from "@angular/core";
-
+import {Component, OnInit, Input, Output, EventEmitter} from "@angular/core";
+import {TestSuiteService} from "./test-suite.service";
+import {TestCaseService} from "../test-cases/test-case.service";
+import {ConfigurationService} from "../configurations/configuration.service";
+import {ToastsManager} from "ng2-toastr/ng2-toastr";
+import {TestCasesSelectionComponent} from "../shared/test-cases-selection/test-cases-selection.component";
 @Component({
   selector: "app-test-suites",
   templateUrl: "./test-suites.component.html",
   styleUrls: ["./test-suites.component.css"]
 })
-export class TestSuitesComponent {}
+export class TestSuitesComponent implements OnInit {
+  testSuitePath = undefined;
+  testCaseArray = [];
+  testSuites = [];
+  selectedTestCaseData = [];
+  testSuiteData = {};
+  testCases = [];
+  testSuiteFileName = undefined;
+  testSuiteFileNameTruncated = undefined;
+  testSuiteSchema = {layout: true};
+  
+  constructor(
+    private testSuiteService: TestSuiteService,
+    private testCaseService: TestCaseService,
+    private configurationService: ConfigurationService,
+    private toastr: ToastsManager
+  ) {}
+
+  ngOnInit() {
+    this.configurationService
+      .getSchema$("assets/testSuite_schema.json")
+      .subscribe((data: any) => {
+        this.testSuiteSchema = data;
+      });
+  }
+
+  onAdd() {
+    this.testSuiteService.getTestSuites$(this.testSuitePath).subscribe(
+      data => {
+        this.testSuites = data;
+      },
+
+      error => {
+        switch (error.status) {
+          case 500: {
+            this.toastr.error("An error has occurred!", "Check the logs!");
+            break;
+          }
+          case 400: {
+            this.toastr.error(
+              "No Test Suite Directory added",
```

Comment:
an error occurred because the request in incorrect. Should we point the user towards looking at the request that was made?